### ISSUE TITLE
Sites not loading in Safari

### ISSUE THREAD

**Issue by sellersshrug0y** [trailofbits/algo#14449]
Algo seems to be preventing the Safari web browser (version 15.4, MacOS 12.3) from loading certain websites including Google search results. I'm using DigitalOcean as my provider, and I've encountered the issue using multiple different Algo instances (some with DNS ad blocking enabled and others disabled.) Instead of loading a webpage, Safari will show an error that reads: "Safari can't open the page. The error is: 'The operation couldn't be completed. Message too long' (NSPOSIXErrorDomain:40)"

The Chrome browser loads the effected webpages without issue. Safari can also load these pages, but only after I've disabled the Algo VPN. 

The expected behavior, of course, is that these webpages load in Safari. I haven't had this issue in the past. I recently updated my computer's OS to MacOS 12.3, and the Algo Safari issue appeared shortly thereafter. It's possible these events are connected.

**Comment by davidemyers** [comment]
Any chance you have Private Relay enabled (in iCloud settings in System Preferences)?

**Comment by sellersshrug0y** [comment]
yep, it was enabled, and disabling it appears to solve the issue. that was easy. thanks!

**Comment by sellersshrug0y** [comment]
actually, i may have spoken too soon. disabling private relay, clearing some cookies, and restarting Safari had appeared to fix the issue, but I've returned to my computer a few hours later and Safari is once again showing an error message instead of loading certain webpages. private relay is still disabled, and i once again cleared some cookies and relaunched the browser. i'm very open to the possibility that this is a "me" problem for one reason or another. i do have a few extensions installed, for instance, and i will try disabling those to see if anything changes. but, again, Algo appears to be the culprit here since Safari works as expected when the VPN is deactivated.

**Comment by jackivanov** [comment]
@sellersshrug0y Did you choose to enable DNS adblocking? I've seen it before with Safari and Argo DNS adblocking enabled. It's related to how Safari handles multiple NXDOMAIN.

**Comment by sellersshrug0y** [comment]
@jackivanov thanks! switching to a Algo instance with DNS adblocking turned off appears the fix the issue. it's still strange, though. I've been using Algo with DNS adblocking for years and haven't experienced this issue until very recently. is it possible something changed in the latest version of Safari that makes this issue more likely to occur?

**Comment by mangajews** [comment]
FWIW, I also experience the same Safari failure albeit under a very different environment.
- I had also found disabling Private Relay as a red herring

I don't use DNS adblocking in Algo, but I do run Adblock on my router which employs the same NXDOMAIN strategy. Safari fails on certain sites with a high probability, although the error message is:
- Safari can’t establish a secure connection to the server "..."

Most interestingly, showing the Web Inspector window and reloading the site will eventually work.

**Comment by cleanbit** [comment]
I have two Algo instances, one instance is having this issue, the other one is not. The cfg is a bit different on non-working instance:
```
BetweenClients_DROP: false
block_smb: false
block_netbios: false
```

Private Relay is disabled.

Is there anything else I could provide while debugging this issue?

**Comment by tarassh** [comment]
The issue also has relation iCloud storage size, when you are running out of space you will get this error.

**Comment by cleanbit** [comment]
Actually, upgrading to macOS 12.5 fixed this issue.

**Comment by mangajews** [comment]
> Actually, upgrading to macOS 12.5 fixed this issue.

Same here.
